msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByAuthorizationGrantSupplier.java

```diff
@@ -39,8 +39,7 @@ AuthenticationResult execute() throws Exception {
         }
 
         if (authGrant instanceof OAuthAuthorizationGrant) {
-            msalRequest.msalAuthorizationGrant =
-                    processPasswordGrant((OAuthAuthorizationGrant) authGrant);
+            processPasswordGrant((OAuthAuthorizationGrant) authGrant);
```

Comment:
It'a bit strange that the authorization grant needs to know about ROPC. Seems like a violation of Single Responsability Principle?